tests/test_measure.py

```diff
@@ -147,3 +147,138 @@ def circuit():
 
         with pytest.raises(QuantumFunctionError, match="CNOT is not an observable"):
             res = circuit()
+
+
+class TestSample:
+    """Tests for the sample function"""
+
+    def test_sample_dimension(self, tol):
+        """Test that the sample function outputs samples of the right size"""
+        dev = qml.device("default.qubit", wires=2)
+
+        n_sample = 10
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(0.54, wires=0)
+            return qml.sample(qml.PauliZ(0), n_sample), qml.sample(qml.PauliX(1), 2*n_sample)
+
+        sample = circuit()
+
+        assert np.array_equal(sample.shape, (2,))
+        assert np.array_equal(sample[0].shape, (n_sample,))
+        assert np.array_equal(sample[1].shape, (2*n_sample,))
+
+    def test_sample_combination(self, tol):
+        """Test the output of combining expval, var and sample"""
+        dev = qml.device("default.qubit", wires=3)
+
+        n_sample = 10
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(0.54, wires=0)
+
+            return qml.sample(qml.PauliZ(0), n_sample), qml.expval(qml.PauliX(1)), qml.var(qml.PauliY(2))
+
+        result = circuit()
+
+        assert np.array_equal(result.shape, (3,))
+        assert np.array_equal(result[0].shape, (n_sample,))
+        assert isinstance(result[1], float)
+        assert isinstance(result[2], float)
+
+    def test_single_wire_sample(self, tol):
+        """Test the return type and shape of sampling a single wire"""
+        dev = qml.device("default.qubit", wires=1)
+
+        n_sample = 10
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.RX(0.54, wires=0)
+
+            return qml.sample(qml.PauliZ(0), n_sample)
+
+        result = circuit()
+
+        assert isinstance(result, np.ndarray)
+        assert np.array_equal(result.shape, (n_sample,))
+
+    def test_multi_wire_sample(self, tol):
+        """Test the return type and shape of sampling multiple wires"""
+        dev = qml.device("default.qubit", wires=3)
+
+        n_sample = 10
+
+        @qml.qnode(dev)
+        def circuit_a():
+            return qml.sample(qml.PauliZ(0), n_sample), qml.sample(qml.PauliZ(1), n_sample), qml.sample(qml.PauliZ(2), n_sample)
+
+        result = circuit_a()
+        print(type(result))
```

Comment:
remember to remove all print statements before the final merge

Reply:
Good catch!